src/microbe_stage/editor/CellEditorComponent.cs

```diff
@@ -784,27 +825,40 @@ public override bool CanFinishEditing(IEnumerable<EditorUserOverride> userOverri
         return true;
     }
 
+    public void SetPatchConditions(Patch patch)
+    {
+        editorPatchConditions = (BiomeConditions)patch.Biome.Clone();
+
+        // If the editor has initialised (i.e. if this is a change of patch during an editor session), switch back to
+        // daytime light levels
+        if (Editor.Ready)
+            SetLightLevelOption("Day");
+    }
+
     public void UpdatePatchDependentBalanceData()
     {
         // Skip if not opened in the multicellular editor
         if (IsMulticellularEditor && editedMicrobeOrganelles.Organelles.Count < 1)
             return;
 
+        topPanel.Visible = Editor.CurrentGame.GameWorld.WorldSettings.DayNightCycleEnabled &&
+            Editor.CurrentPatch.GetCompoundAmount("sunlight", CompoundAmountType.Maximum) > 0.0f;
```

Comment:
sunlight should be retrieved as a Compound object and used here instead of a string key.